tuf/src/database.rs

```diff
@@ -99,21 +100,60 @@ impl<D: DataInterchange> Database<D> {
         })
     }
 
+    /// Create a new [`Database`] struct from a set of metadata that is assumed to be trusted. that
+    /// are used to verify the signed metadata. The signed root metadata must be signed with at
+    /// least a `root_threshold` of the provided root_keys. It is not necessary for the root
+    /// metadata to contain these keys.
```

Comment:
> It is not necessary for the root metadata to contain these keys.

Can you explain why that is? I'd think that the root metadata would have to contain the signing keys.

Reply:
In TUF, we only ever use keys from a trusted root metadata, but we need a way to bootstrap the problem. These keys in effect are acting like the trusted keys from an old trusted root metadata, and they go through the same verification process that we use when doing a root key rotation. We check that the old trusted root keys signed the new root metadata, but once we trust the new root metadata, we only will ever use the new root keys the next time we do an update.

The [spec](https://theupdateframework.github.io/specification/latest/#update-root) goes into a lot of detail for this.

Reply:
Gotcha, thanks.